### PR TITLE
[#270] Override update notifications preferences

### LINKED ISSUE #270: [FEATURE] Override update notifications preferences
**Is your feature request related to a problem? Please describe.**
In the Group we will have another entry for all users called "Notifications". This will yield a panel similar to Group Updates when clicking the Notification icon at the top of the home panel, but it will have an additional checkbox called "Override my notification preferences for this group". See image attached.

group_notification_override

**Describe the solution you'd like**

**Describe alternatives you've considered**

**Additional context**

### PR DESCRIPTION
**Pull request by mdryankov**
## Description

**Resolves #270 **

[comment]: # (This project only accepts pull requests related to open issues.)
[comment]: # (If suggesting a new feature or change, please discuss it in an issue first.)
[comment]: # (If fixing a bug, there should be an issue describing it with steps to reproduce.)

## Review Time Estimate
_Please give your idea of how soon this pull request needs to be reviewed by selecting one of the options below. This can be based on the criticality of the issue at hand and/or other relevant factors._

[comment]: # (To select an option, please put an 'x' in the applicable box.)
[comment]: # (If you're unsure about any of these, don't hesitate to ask. We're here to help!.)

- [x] Immediately
- [ ] Within a week
- [ ] When possible

## Type of changes
_Please select a relevant option:_

[comment]: # (To select an option, please put an 'x' in the applicable box.)
[comment]: # (If you're unsure about any of these, don't hesitate to ask. We're here to help!.)

- [ ] Bug fix (non-breaking change which fixes an issue).
- [x] New feature (non-breaking change which adds functionality).
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected).
- [ ] Other (any another change that does not fall in one of the above categories.)

## Checklist:
_Please select all applicable options:_

[comment]: # (To select your options, please put an 'x' in the all boxes that apply.)
[comment]: # (If you're unsure about any of these, don't hesitate to ask. We're here to help!.)

- [x] I have signed the Rokwire Contributor License Agreement ([CLA](https://rokwire.org/rokwire_cla)). (_Any contributor who is not an employee of the University of Illinois whose official duties include contributing to the Rokwire software, or who is not paid by the Rokwire project, needs to sign the CLA before their contribution can be accepted._)
- [x] I have updated the [CHANGELOG](../CHANGELOG.md).
- [ ] I have read the [Contributor Guidelines](../CONTRIBUTING.md).
- [x] I have performed a self-review of my own code.
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] My change requires updating the documentation.
- [ ] I have made necessary changes to the documentation.
- [ ] I have added tests related to my changes.
- [ ] My changes generate no new warnings.
- [ ] New and existing unit tests pass locally with my changes.
- [ ] Any dependent changes have been merged and published in downstream modules.

---

[comment]: # (Template credit: This pull request template is based on Embedded Artistry {https://github.com/embeddedartistry/templates/blob/master/.github/PULL_REQUEST_TEMPLATE.md}, Clowder {https://github.com/clowder-framework/clowder/blob/develop/.github/PULL_REQUEST_TEMPLATE.md}, and TalAter {https://github.com/TalAter/open-source-templates} templates.)

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -6,6 +6,8 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ## Unreleased
 
+- Override update notifications preferences [#270]((https://github.com/rokwire/groups-building-block/issues/270))
```
Comment (shurwit) [inline CHANGELOG.md:9]: Double parentheses breaks link

File: core/services.go [hunk 2]
```diff
@@ -276,30 +284,38 @@ func (app *Application) createEvent(clientID string, current *model.User, eventI
...
-		result, _ := app.storage.FindGroupMemberships(clientID, model.MembershipFilter{
-			GroupIDs: []string{group.ID},
-			Statuses: []string{"member", "admin"},
-		})
-		recipients = result.GetMembersAsNotificationRecipients(skipUserID)
+		userIDs = []string{group.ID}
+	}
+
+	result, _ := app.storage.FindGroupMemberships(clientID, model.MembershipFilter{
+		GroupIDs: userIDs,
```
Comment (shurwit) [inline core/services.go:296]: Why are we passing `userIDs` in as `GroupIDs`?

File: core/services.go [hunk 3]
```diff
@@ -276,30 +284,38 @@ func (app *Application) createEvent(clientID string, current *model.User, eventI
...
 	if len(event.ToMembersList) > 0 {
-		recipients = event.GetMembersAsNotificationRecipients(skipUserID)
+		userIDs = event.GetMembersAsUserIDs(skipUserID)
 	} else {
-		result, _ := app.storage.FindGroupMemberships(clientID, model.MembershipFilter{
-			GroupIDs: []string{group.ID},
-			Statuses: []string{"member", "admin"},
-		})
-		recipients = result.GetMembersAsNotificationRecipients(skipUserID)
+		userIDs = []string{group.ID}
```
Comment (shurwit) [inline core/services.go:292]: Why are we putting `group.ID` into a variable called `userIDs`?

**Review by shurwit — COMMENTED** [review]
Thanks @mdryankov, I like the strategy for filtering members! I left a few comments below.

File: core/services.go (hunk 2 above)
Comment (mdryankov) [reply core/services.go:296]: Many thanks for the catch. It's a pure cognitive bias without an explanation :)

File: core/services.go (hunk 3 above)
Comment (mdryankov) [reply core/services.go:292]: Same as above

**Pull request merged**